### PR TITLE
Categorical bits and pieces

### PR DESCRIPTION
**Pull request by jpoiret**
Hi,

While attempting a categorical formulation of #1007 that I abandoned in the end, I collected a lot of simple results and modifications.

Noteworthy changes are 2491f5fd4a58ce99f6fdf93c4fd45e08e7b95950, a change to the definition of being an equivalence, which previously was not an hProp. I fixed all uses of it in the library to be compatible with the new definition. I also separated triangle identities from the definition of an adjunction in c55988cfb8a9c66526068fb134bf2547c5c93ef2, so that I could more easily add adjoint equivalences in c9cb16baa0ba94a140b4706a4d418bbe4fe31a21.

From what I can tell this is independent to #988, and shouldn't interfere with it.

LMKWYT

### CODE REVIEW

**Comment by maxsnew** [comment]
This is independent of what's in our current PR and should only have minor conflicts with our followup. Thanks for the heads up it shouldn't be a big issue for us to work around.

File: Cubical/Categories/Presheaf/Constructions.agda [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+module Cubical.Categories.Presheaf.Constructions where
+
+open Functor
+
+module Single
+  {ℓC ℓ'C ℓSet : Level}
+  {C : Category ℓC ℓ'C}
+  where
+
+  constantPresheaf : hSet ℓSet → Presheaf C ℓSet
```
Comment (maxsnew) [inline Cubical/Categories/Presheaf/Constructions.agda:19]: There's already https://github.com/agda/cubical/blob/master/Cubical/Categories/Functors/Constant.agda

Comment (jpoiret) [reply Cubical/Categories/Presheaf/Constructions.agda:19]: Oops, didn't think of looking that way, thanks!

**Comment by jpoiret** [comment]
I just removed the stuff about subpresheaves and representable presheaves, as there's going to be something better with #988 anyways (I have to make time to send some changes to it).

**Comment by felixwellen** [comment]
Can you rebase?

**Review by felixwellen — APPROVED** [review]
Thanks for the contribution!
I commented only on one small detail and will merge anyway if you don't have the time to fix it. 
One question though: Are there any candidates we should consider as a definition of ```isEquivalence```, which are naturally a proposition?

**Comment by felixwellen** [comment]
Thanks! Maybe you can put what you know into a comment at ```isEquivalence```, I guess it is natural to wonder about that when reading the definition.

**Comment by felixwellen** [comment]
With "what you know" I mean that we don't know a notion which works analogous to "isEquiv" for not necessarily univalent categories.

**Comment by felixwellen** [comment]
Thanks!
I'll merge.

**Pull request merged**